gdal/frmts/pds/isis3dataset.cpp

```diff
@@ -3816,338 +3360,340 @@ CPLString ISIS3Dataset::SerializeAsPDL( json_object* poObj )
 /*                      SerializeAsPDL()                                */
 /************************************************************************/
 
-void ISIS3Dataset::SerializeAsPDL( VSILFILE* fp, json_object* poObj,
+void ISIS3Dataset::SerializeAsPDL( VSILFILE* fp, const CPLJSONObject &oObj,
                                    int nDepth )
 {
     CPLString osIndentation;
     for( int i = 0; i < nDepth; i++ )
         osIndentation += "  ";
     const size_t WIDTH = 79;
 
-    json_object_iter it;
-    it.key = nullptr;
-    it.val = nullptr;
-    it.entry = nullptr;
+    CPLJSONObject **paoObjChildren = oObj.GetChildren();
     size_t nMaxKeyLength = 0;
-    json_object_object_foreachC( poObj, it )
+    if( paoObjChildren != nullptr )
     {
-        if( it.val == nullptr ||
-            strcmp(it.key, "_type") == 0 ||
-            strcmp(it.key, "_container_name") == 0 ||
-            strcmp(it.key, "_filename") == 0 )
+        int i = 0;
+        while( paoObjChildren[i] != nullptr )
```

Comment:
couldn't we have C++11 range based loops for this ? There are some tips on how to do this on https://stackoverflow.com/questions/8164567/how-to-make-my-custom-type-to-work-with-range-based-for-loops

Reply:
Do you mean to change 
`CPLJSONObject **GetChildren() const;` 
to 
`std::vector<CPLJSONObject> GetChildren() const;` ?

Reply:
Wasn't thinking to a particular solution. Just pointing to something I had in mind. But the std::vector GetChildren() approach could be a way of solving it

Reply:
Done